### PR TITLE
Add support for nested overrides on connection config

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds support for nested overrides on connection config.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by angrybayblade** [comment]
Although this patch works, we need to keep it in the mind that this is a patch and needs to be addressed properly on open-aea. Right now only skill config has support for nested overrides which should not be the case, I'll open a PR on open-aea to address this later this week.

File: autonomy/configurations/base.py [hunk 1]
```diff
@@ -44,6 +41,19 @@
...
+    FIELDS_WITH_NESTED_FIELDS: FrozenSet[str] = frozenset(
+        [
+            "config",
+        ]
+    )
+    NESTED_FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset(
+        ["ledger_apis", "ethereum"]
+    )
+
+
```
Comment (8ball030) [inline autonomy/configurations/base.py:56]: How come this was not needed for models?

File: autonomy/configurations/base.py [hunk 2]
```diff
@@ -366,12 +366,6 @@ def try_to_process_nested_fields(  # pylint: disable=too-many-locals
                         nested_override_key,
                         nested_override_value,
                     ) in nested_value.items():
-                        if (
-                            nested_override_key
-                            not in config_class.NESTED_FIELDS_ALLOWED_TO_UPDATE  # type: ignore
-                        ):
-                            raise ValueError("Trying to override non-nested field.")
```
Comment (angrybayblade) [inline autonomy/configurations/base.py:373]: We don't need to check for `NESTED_FIELDS_ALLOWED_TO_UPDATE`, we're planning to remove this variable in one of the coming releases

**Review by Adamantios — COMMENTED** [review]
Can we add a test for this?

File: autonomy/configurations/base.py [hunk 3]
```diff
@@ -391,4 +385,5 @@ def try_to_process_nested_fields(  # pylint: disable=too-many-locals
 PACKAGE_TYPE_TO_CONFIG_CLASS = {
     **_PACKAGE_TYPE_TO_CONFIG_CLASS,
     PackageType.SERVICE: Service,
+    PackageType.CONNECTION: ConnectionConfig,
```
Comment (DavidMinarsch) [inline autonomy/configurations/base.py:388]: I think this can go. `PackageType.CONNECTION` is already specified in `_PACKAGE_TYPE_TO_CONFIG_CLASS`

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: autonomy/configurations/base.py (hunk 3 above)
Comment (angrybayblade) [reply autonomy/configurations/base.py]: https://github.com/valory-xyz/open-aea/pull/298/files#diff-9d90463edcd0573846e5906abe4e7def07e18ad71ed8b7ae1da9f8e33330ec4bR33-R39

File: tests/test_base.py [hunk 4]
```diff
@@ -87,9 +88,10 @@ def test_load_all_packages(component_type: PackageType, config_file_path: str) -
     )
     with open_file(config_file_path) as fp:
         configuration_object = configuration_loader.load(fp)
-        directory = Path(config_file_path).parent
-        configuration_object.directory = directory
-        load_aea_package(configuration_object)
+        if component_type != PackageType.SERVICE:
+            directory = Path(config_file_path).parent
+            configuration_object.directory = directory
+            load_aea_package(configuration_object)
```
Comment (DavidMinarsch) [inline tests/test_base.py:94]: I think this line should not be nested?

Why are services not being assigned a directory?

Comment (angrybayblade) [reply tests/test_base.py:94]: Sorry, for not providing context. The nested logic also loads dependency packages which is not required in service component. We only want to check if we can load a service component config without any error.

Comment (DavidMinarsch) [reply tests/test_base.py:94]: This comment should really be there in code. I will merge into the base branch as you need to fix things there but please add it.

**Pull request merged**